rllib/utils/minibatch_utils.py

```diff
@@ -9,75 +9,94 @@
+                in multi-agent + multi-GPU situations, in which the MultiAgentEpisodes
+                themselves are roughly sharded equally, however, they might contain
+                SingleAgentEpisodes with very lopsided length distributions. Thus,
+                without this fixed, pre-computed value, one Learner might go through a
+                different number of minibatche passes than others causing a deadlock.
+        """
         pass
 
 
 @DeveloperAPI
 class MiniBatchCyclicIterator(MiniBatchIteratorBase):
     """This implements a simple multi-agent minibatch iterator.
 
-
     This iterator will split the input multi-agent batch into minibatches where the
     size of batch for each module_id (aka policy_id) is equal to minibatch_size. If the
     input batch is smaller than minibatch_size, then the iterator will cycle through
-    the batch until it has covered num_iters epochs.
-
-    Args:
-        batch: The input multi-agent batch.
-        minibatch_size: The size of the minibatch for each module_id.
-        num_iters: The minimum number of epochs to cover. If the input batch is smaller
-            than minibatch_size, then the iterator will cycle through the batch until
-            it has covered at least num_iters epochs.
+    the batch until it has covered `num_epochs` epochs.
     """
 
     def __init__(
         self,
         batch: MultiAgentBatch,
+        *,
+        num_epochs: int = 1,
         minibatch_size: int,
-        num_iters: int = 1,
-        uses_new_env_runners: bool = False,
-        num_total_mini_batches: int = 0,
+        shuffle_batch_per_epoch: bool = True,
+        num_total_minibatches: int = 0,
+        _uses_new_env_runners: bool = False,
     ) -> None:
-        super().__init__(batch, minibatch_size, num_iters)
+        """Initializes a MiniBatchCyclicIterator instance."""
+        super().__init__(
+            batch,
+            num_epochs=num_epochs,
+            minibatch_size=minibatch_size,
+            shuffle_batch_per_epoch=shuffle_batch_per_epoch,
+        )
+
         self._batch = batch
         self._minibatch_size = minibatch_size
-        self._num_iters = num_iters
+        self._num_epochs = num_epochs
+        self._shuffle_batch_per_epoch = shuffle_batch_per_epoch
 
         # mapping from module_id to the start index of the batch
         self._start = {mid: 0 for mid in batch.policy_batches.keys()}
         # mapping from module_id to the number of epochs covered for each module_id
         self._num_covered_epochs = {mid: 0 for mid in batch.policy_batches.keys()}
 
-        self._uses_new_env_runners = uses_new_env_runners
+        self._uses_new_env_runners = _uses_new_env_runners
 
-        self._mini_batch_count = 0
-        self._num_total_mini_batches = num_total_mini_batches
+        self._minibatch_count = 0
+        self._num_total_minibatches = num_total_minibatches
 
     def __iter__(self):
```

Comment:
In the long run we might want to override `DataIterator` from `ray.data` to build batches from `MultiAgentEpisode`s. Less code.